tests/test_conversion.py

```diff
@@ -720,3 +720,17 @@ def test_intptr_construction():
         with pytest.raises(OverflowError):
             UIntPtr(v)
 
+def test_explicit_conversion():
+    from System import (
+        Int64, UInt64, Int32, UInt32, Int16, UInt16, Byte, SByte, Boolean
+    )
+    from System import Double, Single
+
+    assert int(Boolean(True)) == 1
+
+    for t in [Int64, UInt64, Int32, UInt32, Int16, UInt16, Byte, SByte]:
+        assert int(t(127)) == 127
+        assert float(t(127)) == 127.0
```

Comment:
Try edge cases (`UInt64.MaxValue`, `Int64.MinValue`) in addition to regular values